src/core/functions/table/match.cpp

```diff
@@ -82,6 +82,61 @@ case_insensitive_set_t GetFullyQualifiedColFromPg(
   return col_names;
 }
 
+// Get all fully-qualified column names from the given property graph [pg] for
+// the given relation [alias], only vertex table is selected.
+//
+// Return a vector of column names, each of them represents a stack of names in
+// order of which they appear
+// (column_names[0].column_names[1].column_names[2]....).
+vector<vector<string>> GetRegisteredColFromPg(
+    const case_insensitive_map_t<shared_ptr<PropertyGraphTable>> &alias_map,
+    const std::string &alias) {
+  vector<vector<string>> registered_col_names;
+  auto iter = alias_map.find(alias);
+  D_ASSERT(iter != alias_map.end());
+  const auto &tbl = iter->second;
+  // Skip edge table.
+  if (!tbl->is_vertex_table) {
+    return registered_col_names;
+  }
+  registered_col_names.reserve(tbl->column_names.size());
+  for (const auto &cur_col : tbl->column_names) {
+    registered_col_names.emplace_back(vector<string>{"", ""});
+    auto &new_col_names = registered_col_names.back();
+    new_col_names[0] = alias;
+    new_col_names[1] = cur_col;
+  }
+  return registered_col_names;
+}
+
+// Get all fully-qualified column names from the given property graph [pg] for
+// all vertex relations.
+//
+// Return a vector of column names, each of them represents a stack of names in
+// order of which they appear
+// (column_names[0].column_names[1].column_names[2]....).
+vector<vector<string>> GetRegisteredColFromPg(
+    const case_insensitive_map_t<shared_ptr<PropertyGraphTable>> &alias_map) {
+  vector<vector<string>> registered_col_names;
```

Comment:
I think it would be better if this GetRegisteredColFromPG would return a vector of uniq_ptr<ColumnRefExpressions> instead of vector of strings. That should make it clearer what we are doing here.

Reply:
sure, sounds good, updated